metadata-io/src/main/java/com/linkedin/metadata/entity/EntityService.java

```diff
@@ -1522,6 +1522,12 @@ private Map<EntityAspectIdentifier, EnvelopedAspect> getEnvelopedAspects(final S
       //    since nowhere else is using it should be safe for now at least
       envelopedAspect.setType(AspectType.VERSIONED);
       envelopedAspect.setValue(aspect);
+
+      if (currAspectEntry.getSystemMetadata() != null) {
+        final SystemMetadata systemMetadata = RecordUtils.toRecordTemplate(SystemMetadata.class, currAspectEntry.getSystemMetadata());
```

Comment:
in case people have b0rked systemMetadata entries, surround this in a try catch with default exception path being `log.warn` and don't set systemMetadata.

Reply:
not a bad call...

Reply:
nice